presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1457,6 +1456,69 @@ protected Node aggregateResult(Node aggregate, Node nextResult)
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    private static String decodeStringExpression(SqlBaseParser.StringContext context)
+    {
+        String rawContent = unquote(context.STRING().getText());
+        if (context.ESCAPE_INTRODUCER() == null) {
+            return rawContent;
+        }
+        String result = "";
+        String currentEscapedCode = "";
+        int lengthBeforeComplete = 0;
+        for (int i = 0; i < rawContent.length(); i++) {
+            if (rawContent.charAt(i) == '\\') {
+                if (lengthBeforeComplete != 0) {
+                    throw parseError("Incomplete escaped character: " + currentEscapedCode, context);
+                }
+                if (i < rawContent.length() - 1 && rawContent.charAt(i + 1) == '+') {
+                    lengthBeforeComplete = 6;
+                    i++;
+                }
+                else {
```

Comment:
This also needs to handle escaping the escape character by doubling it. See `<Unicode character escape value>` in the specification grammar.